### PR TITLE
(feat): support WDIO v9

### LINKED ISSUE #533: Update service for WDIO v9
WDIO v9 is coming along nicely, there is a bit of work to do in the service around this, since we are dropping Node 16 I think we cut a major version and punt the IPC bridge replacement to v8.

Tasks:
- [x] Publish working pre-release builds 
- [x] Ensure dev docs are up to date
  - Document the fix in https://github.com/webdriverio-community/wdio-electron-service/discussions/667
- [x] Manual test with `wdio-electron-service-example`
- [ ] Manual test with `electron-boilerplate` (can use the [v9 branch](https://github.com/webdriverio/electron-boilerplate/tree/v9))
- [x] Update WDIO peerdep range to ">9.0.0" (after the WDIO v9 release)
- [x] Update WDIO deps to "^9.0.0" (after the WDIO v9 release)
- [ ] Publish full release

https://github.com/webdriverio/webdriverio/issues/10362

### PR DESCRIPTION
**Pull request by goosewobbler**
Closes #533.  Drops support for N16, The minimum Node version is now 18.

TODO:

- [x] Ensure types are as accurate as possible
- [x] Drop support for N16
- [x] Remove `node-fetch`, use native `fetch`
- [x] Remove `ts-node`
- [x] Investigate removal of `ts-node` dynamic import issue workaround 
  - successfully removed from the E2Es
  - removing it from the CJS index [breaks the CJS build](https://github.com/webdriverio-community/wdio-electron-service/actions/runs/9861121162/job/27228861273?pr=666) 
    - CJS build approach likely needs reworking
    - leaving it for now with an updated comment

### CODE REVIEW

**Comment by christian-bromann** [comment]
Thanks for helping getting v9 adopted in our ecosystem projects. Please raise an issue in the main repo if you see any strange things.

**Comment by goosewobbler** [comment]
I'm not sure about some of the type changes.  I don't think it's far off, but I've just added an `any` to fix the build...

File: packages/wdio-electron-service/src/launcher.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ export default class ElectronLaunchService implements Services.ServiceInstance {
   async onPrepare(_config: Options.Testrunner, capabilities: ElectronServiceCapabilities) {
     const capsList = Array.isArray(capabilities)
       ? capabilities
-      : Object.values(capabilities).map((multiremoteOption) => multiremoteOption.capabilities);
+      : Object.values(capabilities).map((multiremoteOption) => (multiremoteOption as any).capabilities);
```
Comment (christian-bromann) [inline packages/wdio-electron-service/src/launcher.ts:30]:
```suggestion
      : Object.values(capabilities as Capabilities.RequestedMultiremoteCapabilities).map((multiremoteOption) => (multiremoteOption as Capabilities.WithRequestedCapabilities).capabilities);
```

File: packages/wdio-electron-service/src/capabilities.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ const isElectron = (cap: unknown) => (cap as WebdriverIO.Capabilities)?.browserN
 /**
  * Get capability independent of which type of capabilities is set
  */
-export function getElectronCapabilities(caps: Capabilities.RemoteCapability) {
+export function getElectronCapabilities(caps: Capabilities.TestrunnerCapabilities) {
```
Comment (christian-bromann) [inline packages/wdio-electron-service/src/capabilities.ts:25]: Given that we already abstract caps from possible multiremote capabilities, this type should be
```suggestion
export function getElectronCapabilities(caps: Capabilities.RequestedStandaloneCapabilities) {
```

File: packages/wdio-electron-service/src/capabilities.ts [hunk 3]
```diff
@@ -45,7 +45,7 @@ export function getElectronCapabilities(caps: Capabilities.RemoteCapability) {
    * }
    * ```
    */
-  const w3cCaps = (caps as Capabilities.W3CCapabilities).alwaysMatch;
+  const w3cCaps = (caps as unknown as Capabilities.W3CCapabilities).alwaysMatch;
```
Comment (christian-bromann) [inline packages/wdio-electron-service/src/capabilities.ts:48]:
```suggestion
  const w3cCaps = (caps as Capabilities.W3CCapabilities).alwaysMatch;
```

**Review by christian-bromann — COMMENTED** [review]
Some type suggestions

**Review by christian-bromann — APPROVED** [review]
🚀

**Pull request merged**